### DISCUSSION TITLE
Visualisasi - R - Bagaimana cara memvisualisasi 1 vector dengan 2 class berbeda di dalamnya, bersama vector kolom lain pada dataframe.

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#69] [Q&A] [answered]
1. Saya melakukan imputation dengan dlookr package, imputate_na(), yang menghasilkan class "imputation" pada suatu vector dengan class "factor".
2. Dari dlookr packagenya sendiri sebenarnya sudah menyediakan visualisasi pembanding antara vector original dan vector hasil imputation. Namun jika saya menginsert vector imputation ke dataframe asal, dan ingin melakukan visualisasi vector tersebut bersama variabel lain di dataset, namun tetap ingin memberi pembeda berdasarkan class berbeda pada 1 vector variabel tersebut bagaimana kiranya caranya?

Kalau ditanya ini dataset apa, sebenarnya ini dataset kiva yang dari quiz DV, dan vector yang dibahas adalah vector kolom region.

**Comment by NabiilahArdini** [comment, marked as answer]
Halo Bapak Husada @Hertzspr, ada contoh datanya kah Pak? Boleh disertakan mungkin code untuk proses imputasi dengan dlookr package nya. Supaya bisa kita coba juga.

**Reply by Hertzspr** [reply]
Halo Mbak Nabiilah, ini dataset kiva dari yang [quiz DV](https://algoritmaonline.com/courses/data-visualization-in-r/?passes=1&quiz=14)

Walau ternyata mungkin quiznya bisa selese tanpa imputasi, saya waktu ngerjain sempet kelamaan utak atik imputasi, penasaran, dan sekalian pengen nyoba package dlookr.

```
kiva_reg_imputed <- imputate_na(kiva,
                                region,
                                method="mode"
                                )

kiva$region <- kiva_reg_imputed
```
Nah kalau dicek classnya
`class(kiva$region)`
"imputation" "factor" 

-----

Kalau mau memvisualisasi  kolom region, bareng kolom lain, tapi bisa kelihatan beda antara data region yang "factor" dan data region yang "imputation" gimana ya? Apakah dibuat kolom baru sebagai marker? lalu mengambil hasil return dari fungsi class()?

**Reply by NabiilahArdini** [reply]
Halo Bapak Husada @Hertzspr,

Mohon maaf atas respon yang cukup lama. Kami sudah coba eksplor hasil imputasi tersebut dan belum menemukan cara untuk bisa langsung menvisualisasikannya terpisah antara hasil imputasi dan nilai asli di plot base/ggplot. Cara yang dapat dilakukan adalah membuat 1 kolom baru untuk penanda bahwa terdapat imputasi untuk baris tertentu. 

Kami juga coba membuat data dummy karena data dari kuis tidak memiliki NA. Berikut tahapannya:

```
temp <- data.frame(region = factor(c("Asia", "Asia", "Europe", NA, "Asia")), 
           value = c(109,100,120,114, 115))

# create column for color code
temp$color_code <- ifelse(is.na(temp$region), "imputed", "real")

# impute
temp$region <- dlookr::imputate_na(temp, region,
                                method = "mode")

temp
```
<img width="272" alt="image" src="https://user-images.githubusercontent.com/51941102/167092939-29ceac19-303a-40db-881c-77caca8683d6.png">

```
library(ggplot2)
ggplot(temp, aes(x = region)) +
  geom_bar(aes(fill = color_code))
```
<img width="342" alt="image" src="https://user-images.githubusercontent.com/51941102/167092864-da30a616-793c-4be2-a823-67065892e106.png">


Anda dapat mengatur lebih lanjut untuk kebutuhan visualisasi lainnya. 
Semoga dapat membantu.
